contracts/JBTieredNFTRewardDelegate.sol

```diff
@@ -0,0 +1,382 @@
+
+    // If there's a token to mint, do so and increment the tier supply.
+    _mint(_beneficiary, tokenId);
+
+    emit Mint(tokenId, _tierNumber, _beneficiary, _value, msg.sender);
+  }
+
+  /** 
+    @notice
+    Burns a token ID from an account.
+
+    @dev
+    Only a project owner can burn tokens.
+
+    @param _owner The address that owns the token being burned.
+    @param _tokenId The ID of the token to burn.
+  */
+  function burn(address _owner, uint256 _tokenId) external override onlyOwner {
+    if (_ownerOf[_tokenId] != _owner) revert INCORRECT_OWNER();
+
+    // The token to which the token ID belongs.
+    uint256 _tierNumber = tierNumberOfToken(_tokenId);
+
+    // Burn the token and decrease the supply.
+    _burn(_tokenId);
+
+    // Add to the remaining allowance.
+    unchecked {
+      // The tier number is 1 indexed.
+      ++_tiers[_tierNumber - 1].remainingAllowance;
+    }
+
+    emit Burn(_tokenId, _owner, msg.sender);
+  }
+
+  //*********************************************************************//
+  // ------------------------ internal functions ----------------------- //
+  //*********************************************************************//
+
+  /**
+    @notice 
+    Mints a token for a given contribution to the beneficiary.
+
+    @param _data The Juicebox standard project contribution data.
+  */
+  function _processContribution(JBPayParamsData calldata _data) internal override {
+    // Make the contribution is being made in the expected token.
+    if (_data.amount.token != contributionToken) return;
+
+    // Keep a reference to the token ID.
+    (uint256 _tokenId, uint256 _tierNumber) = _generateTokenId(_data.amount.value);
+
+    // Make sure there's a token ID.
+    if (_tokenId == 0) revert NOT_AVAILABLE();
+
+    // If there's a token to mint, do so and increment the tier supply.
+    _mint(_data.beneficiary, _tokenId);
+
+    emit Mint(_tokenId, _tierNumber, _data.beneficiary, _data.amount.value, msg.sender);
+  }
+
+  /** 
+    @notice
+    Finds the token ID and tier given a contribution amount. 
+
+    @param _amount The amount of the contribution.
+
+    @return The ID of the token.
+    @return The tier number.
+  */
+  function _generateTokenId(uint256 _amount) internal returns (uint256, uint256) {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = _tiers.length;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier memory _tier;
+
+    for (uint256 _i; _i < _numTiers; ) {
+      // Set the tier being iterated on.
+      _tier = _tiers[_i];
+
+      // If the contribution value is at least as much as the floor
+      if (
+        _tier.contributionFloor <= _amount &&
+        (_i == _numTiers - 1 || _tiers[_i + 1].contributionFloor > _amount) &&
+        _tier.remainingAllowance != 0
+      ) {
+        // The token ID incrementally increases until the id cieling.
+        uint256 _tokenId = _tier.idCeiling - _tier.remainingAllowance;
```

Comment:
This prevent the whole first tier being minted -> ceiling-remaining = 0, then the tokenId==0 check in mint(..) or in _processContribution(..) reverts

Reply:
true. id ceil must be greater than initial allowance.